mlx/utils.h

```diff
@@ -9,6 +9,27 @@
 
 namespace mlx::core {
 
+using StreamOrDevice = std::variant<std::monostate, Stream, Device>;
```

Comment:
The header `<variant>` is needed for gcc to compile this. Also the same explicit header include should be removed from `ops.h` now unless it `std::variant` is used elsewhere.

(Didn't see the failed linux build when I did the initial review hence the separate comment)